Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM python:3.9-slim-buster as base
+WORKDIR /todo_app
+RUN pip3 install poetry
+COPY poetry.lock pyproject.toml wsgi.py ./
+RUN poetry config virtualenvs.create false && poetry install
```

Comment:
One option you could consider is splitting this installation out into the two stages separately; you're taking advantage of the idea of Dev dependencies in your `pyproject.toml` file, so your production container could add the `--no-dev` flag to the `poetry install` to avoid pulling those dependencies, which will be a minor optimization but can improve speed/image size and also cuts out a possible avenue of security concerns (if one of your dev libraries is flagged as insecure for example)